client/app/scripts/superdesk-ingest/module.js

```diff
@@ -864,118 +897,95 @@ define([
-
-                scope.genres = [];
-                scope.genreTerm = '';
-                scope.filteredGenres = [];
-
-                subjectService
-                .initialize()
-                .then(function(subjects) {
-                    scope.subjects = subjects;
-                });
-
-                api.get('vocabularies')
-                .then(function(result) {
-                    scope.categories = _.find(result._items, {_id: 'categories'}).items;
-                    scope.genres = _.find(result._items, {_id: 'genre'}).items;
-                });
-
-                scope.isTypeChecked = function(rule, type) {
-                    return rule.filter.type.indexOf(type) !== -1;
-                };
-
-                scope.toggleType = function(rule, type) {
-                    if (scope.isTypeChecked(rule, type)) {
-                        rule.filter.type = _.without(rule.filter.type, type);
-                    } else {
-                        rule.filter.type.push(type);
-                    }
-                };
-
-                scope.removeSubject = function(subject) {
-                    _.remove(scope.rule.filter.subject, function(s) {
-                        return s.qcode === subject.qcode;
-                    });
-                };
-
-                scope.selectSubject = function(item) {
-                    scope.rule.filter.subject.push({qcode: item.qcode, name: item.name});
-                    scope.subjectTerm = '';
-                };
+    /**
+     * @memberof superdesk.ingest
+     * @ngdoc directive
+     * @name sdIngestRoutingFilter
+     * @description
+     *   Creates the Filter tab used for defining a content filter for routing
+     *   rules (found in the modal for editing ingest routing schemes).
+     */
+    IngestRoutingFilter.$inject = [];
+    function IngestRoutingFilter() {
 
-                scope.searchSubjects = function(term) {
-                    var regex = new RegExp(term, 'i');
-                    scope.filteredSubjects = _.filter(scope.subjects, function(subject) {
-                        return (
-                            regex.test(subject.name) &&
-                            _.findIndex(scope.rule.filter.subject, {qcode: subject.qcode}) === -1
-                        );
-                    });
-                };
+        /**
+         * Creates an utility method on the built-in RegExp object used for
+         * escaping arbitrary strings so that they can be safely used in
+         * dynamically created regular expressions patterns.
+         *
+         * Taken from http://stackoverflow.com/a/3561711/5040035
+         *
+         * @method escape
+         * @param {string} s - the string to escape
+         * @return {string} - an escaped version of the given string
+         */
+        // XXX: should probably be moved into some utils module - but where?
+        RegExp.escape = function (s) {
+            return s.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&');
```

Comment:
Some usage examples an expected output will help in reviewing/validating the correctness of the regex.